app/src/behavior.c

```diff
@@ -56,29 +58,58 @@ const struct device *z_impl_behavior_get_binding(const char *name) {
     return NULL;
 }
 
+// TODO: Delete this method as part of a breaking release
+int zmk_behavior_invoke_binding(const struct zmk_behavior_binding *src_binding,
+                                struct zmk_behavior_binding_event event, bool pressed) {
+    LOG_DBG("`zmk_behavior_invoke_binding` is deprecated. Please raise a "
+            "`zmk_behavior_binding_event` instead.");
+    return raise_zmk_behavior_binding_event((struct zmk_behavior_binding_event){
+        .binding = src_binding,
+        .layer = event.layer,
+        .position = event.position,
+        .timestamp = event.timestamp,
+        .type = pressed ? PRESS : RELEASE,
+#if IS_ENABLED(CONFIG_ZMK_SPLIT)
+        .source = event.source,
+#endif
+    });
+}
+
+// TODO: Pass the event in as pointers, rather than copying in the struct
+// Make this change as part of a breaking release
```

Comment:
Isn't this passing it as a pointer already?

Reply:
Got it, so the comment applies to the function calls inside. Though I don't know enough to say whether passing a pointer rather than a struct gives a non-trivial performance gain. I'd be interested to learn if e.g. @joelspadin has thoughts on it.